src/app/vault/share.component.ts

```diff
@@ -69,4 +74,25 @@ export class ShareComponent extends BaseShareComponent {
         } catch { }
         return false;
     }
+
+    get canSave() {
+        const isCipherInOrganization = !!this.cipher.organizationId;
+        const hasSelectedCollection = this.collections.some(c => !!(c as any).checked);
+
+        return hasSelectedCollection || isCipherInOrganization;
+    }
+
+    async unshare(): Promise<boolean> {
+        try {
+            this.formPromise = this.localCipherService.unshare(this.cipher)
+                .then(async () => {
+                    this.onSharedCipher.emit();
+                    this.platformUtilsService.showToast('success', null, this.i18nService.t('unsharedItem'));
+                });
+
+            await this.formPromise;
```

Comment:
what is the benefit to store this formPromise in... `this`? Why not a scoped variable? 

const formPromise = ... ?

Reply:
This is used to make the UI reacting to the validation progress. Here formPromise is refered on the HTML : https://github.com/cozy/cozy-pass-web/pull/44/files#diff-8493f2f471b027fe26312639683dfd8574b89d91763582a76649958654121f04R3

This mechanism is used on all other bitwarden's dialogs, so I did not try to improve it (also I didn't find any cons yet to this form)